presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/common/Hadoop.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.tests.product.launcher.env.common;
+
+import io.prestosql.tests.product.launcher.docker.DockerFiles;
+import io.prestosql.tests.product.launcher.env.DockerContainer;
+import io.prestosql.tests.product.launcher.env.Environment;
+import io.prestosql.tests.product.launcher.env.EnvironmentOptions;
+import io.prestosql.tests.product.launcher.testcontainers.SelectedPortWaitStrategy;
+import org.testcontainers.containers.startupcheck.IsRunningStartupCheckStrategy;
+
+import javax.inject.Inject;
+
+import java.time.Duration;
+
+import static io.prestosql.tests.product.launcher.testcontainers.TestcontainersUtil.exposePort;
+import static java.util.Objects.requireNonNull;
+import static org.testcontainers.containers.BindMode.READ_ONLY;
+
+public final class Hadoop
+        implements EnvironmentExtender
+{
+    public static final String CONTAINER_PRESTO_HIVE_PROPERTIES = "/docker/presto-product-tests/conf/presto/etc/catalog/hive.properties";
+
+    private final DockerFiles dockerFiles;
+
+    private final String hadoopBaseImage;
+    private final String imagesVersion;
+
+    @Inject
+    public Hadoop(
+            DockerFiles dockerFiles,
+            EnvironmentOptions environmentOptions)
+    {
+        this.dockerFiles = requireNonNull(dockerFiles, "dockerFiles is null");
+        requireNonNull(environmentOptions, "environmentOptions is null");
+        hadoopBaseImage = requireNonNull(environmentOptions.hadoopBaseImage, "environmentOptions.hadoopBaseImage is null");
+        imagesVersion = requireNonNull(environmentOptions.imagesVersion, "environmentOptions.imagesVersion is null");
+    }
+
+    @Override
+    public void extendEnvironment(Environment.Builder builder)
+    {
+        builder.addContainer("hadoop-master", createHadoopMaster());
+
+        builder.configureContainer("presto-master", container -> container
+                .withFileSystemBind(dockerFiles.getDockerFilesHostPath("common/hadoop/hive.properties"), CONTAINER_PRESTO_HIVE_PROPERTIES, READ_ONLY));
```

Comment:
> common/hadoop/hive.properties"

why not just `hadoop` or `env/hadoop` or `environment/hadoop`?

Reply:
> > common/hadoop/hive.properties"
> 
> why not just `hadoop` or `env/hadoop` or `environment/hadoop`?

convention

```
- common - this is where all common (reusable env pieces) parts live
  \- hadoop - this is where Hadoop's pieces live
     \- hive.properties - (obvious meaning)
```

i use `common` rather than `env` or `environment` because `hadoop` is not an environment.
It's a reusable piece (like a lib code, utility class).
Environments are separate and not meant to be reused.